client/packages/programs/src/JsonForms/useDocumentFormData.ts

```diff
@@ -68,18 +72,23 @@ export const useDocumentLoader = (
     }
   }, [databaseResponse]);
 
-  // user createDoc if there is one
+  // use schema from the formInputData if there is one
   useEffect(() => {
-    if (createDoc) {
-      setSchema(createDoc.schema);
+    if (formInputData) {
+      setSchema(formInputData.schema);
     }
-  }, [createDoc]);
+  }, [formInputData]);
 
   return {
     isLoading,
     error,
-    documentId,
-    data: createDoc?.data ?? databaseResponse?.data,
+    loadedData: formInputData?.data ?? databaseResponse?.data,
+    isCreating: formInputData?.isCreating ?? false,
     schema,
+    save: handleSave
```

Comment:
When would this not have save method ?

Reply:
No, probably not but will leave for now...